terratest/utils/extensions.go

```diff
@@ -0,0 +1,282 @@
+	}
+}
+
+func (i *Instance) GetIngressIPs() []string {
+	var ingressIPs []string
+	ingress := k8s.GetIngress(i.w.t, i.w.k8sOptions, i.w.settings.ingressName)
+	for _, ip := range ingress.Status.LoadBalancer.Ingress {
+		ingressIPs = append(ingressIPs, ip.IP)
+	}
+	return ingressIPs
+}
+
+func (i *Instance) StopTestApp() {
+	k8s.RunKubectl(i.w.t, i.w.k8sOptions, "scale", "deploy", i.w.state.testApp.name, "--replicas=0")
+	AssertGslbStatus(i.w.t, i.w.k8sOptions, i.w.state.gslb.name, i.w.state.gslb.host+":Unhealthy")
+	i.w.state.testApp.isRunning = false
+}
+
+func (i *Instance) StartTestApp() {
+	k8s.RunKubectl(i.w.t, i.w.k8sOptions, "scale", "deploy", i.w.state.testApp.name, "--replicas=1")
+	AssertGslbStatus(i.w.t, i.w.k8sOptions, i.w.state.gslb.name, i.w.state.gslb.host+":Healthy")
+	i.w.state.testApp.isRunning = true
+}
+
+// WaitForGSLB waits until GSLB contains desired IP address list and if it is, the desired list is returned.
+// Desired IP list is LocalTargetsIP combination of all instances with running app.
+// e.g.: instance1.WaitForGSLB(instance2, instance3) produces:
+// desiredIPList := instance1.GetLocalTargets() + instance2.GetLocalTargets() + instance3.GetLocalTargets()
+// If app is stopped, the IP addresses are excluded from desired list.
+func (i *Instance) WaitForGSLB(instances ...*Instance) ([]string, error) {
+	var expectedIPs []string
+	instances = append(instances, i)
+	for _, in := range instances {
+		// add expected IP's only if app is running
+		if in.w.state.testApp.isRunning {
+			ip := in.GetLocalTargets()
+			expectedIPs = append(expectedIPs, ip...)
+		}
+	}
+	return waitForLocalGSLBNew(i.w.t, i.w.state.gslb.host, i.w.state.gslb.port, expectedIPs)
+}
+
+// WaitForExpected waits until GSLB dig doesnt return list of expected IP's
+func (i *Instance) WaitForExpected(expectedIPs []string) ([]string, error) {
+	return waitForLocalGSLBNew(i.w.t, i.w.state.gslb.host, i.w.state.gslb.port, expectedIPs)
+}
+
+func (i *Instance) String() string {
+	return fmt.Sprintf("Instance: %s", i.w.namespace)
+}
+
+// Dig  returns a list of IP addresses from CoreDNS that belong to the instance
+func (i *Instance) Dig() []string {
+	dig, err := dns.Dig("localhost:"+strconv.Itoa(i.w.state.gslb.port), i.w.state.gslb.host)
+	require.NoError(i.w.t, err)
+	return dig
+}
+
+// GetLocalTargets returns instance local targets
+func (i *Instance) GetLocalTargets() []string {
+	dnsName := fmt.Sprintf("localtargets-%s", i.w.state.gslb.host)
+	dig, err := dns.Dig("localhost:"+strconv.Itoa(i.w.state.gslb.port), dnsName)
+	require.NoError(i.w.t, err)
+	return dig
+}
+
+func waitForLocalGSLBNew(t *testing.T, host string, port int, expectedResult []string) (output []string, err error) {
+	return DoWithRetryWaitingForValueE(
+		t,
+		"Wait for failover to happen and coredns to pickup new values...",
+		100,
+		time.Second*1,
+		func() ([]string, error) { return dns.Dig("localhost:"+strconv.Itoa(port), host) },
+		expectedResult)
+}
```

Comment:
@kuritka overall it looks great. Can we document the testing DLS functions in this file and/or create short documentation?